language/reserved_keywords.rb

```diff
@@ -0,0 +1,155 @@
+require_relative '../spec_helper'
+
+describe "Ruby's reserved keywords" do
+  # Copied from Prism::Translation::Ripper
+  keywords = [
+    "alias",
+    "and",
+    "begin",
+    "BEGIN",
+    "break",
+    "case",
+    "class",
+    "def",
+    "defined?",
+    "do",
+    "else",
+    "elsif",
+    "end",
+    "END",
+    "ensure",
+    "false",
+    "for",
+    "if",
+    "in",
+    "module",
+    "next",
+    "nil",
+    "not",
+    "or",
+    "redo",
+    "rescue",
+    "retry",
+    "return",
+    "self",
+    "super",
+    "then",
+    "true",
+    "undef",
+    "unless",
+    "until",
+    "when",
+    "while",
+    "yield",
+    "__ENCODING__",
+    "__FILE__",
+    "__LINE__"
+  ]
+
+  keywords.each do |kw|
+    describe "keyword '#{kw}'" do
+      it "can't be used as local variable name" do
+        expect_syntax_error <<~RUBY
+            #{kw} = "a local variable named '#{kw}'"
+        RUBY
+      end
+
+      invalid_ivar_names = ["defined?"]
+
+      if invalid_ivar_names.include?(kw)
```

Comment:
`if` outside examples like this are pretty much an anti-pattern in ruby/spec.
Same for looping around `it`.
Instead, the loops should be inside the `it`.
That also makes it easier to read, e.g. `(keywords - %w[defined?]).each`.